src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -3332,7 +3358,13 @@ MultiRouterPlannableQuery(Query *query)
 			/* local tables are allowed if there are no distributed tables */
 			if (!IsCitusTable(distributedTableId))
 			{
-				hasLocalTable = true;
+				hasPostgresLocalTable = true;
+				continue;
+			}
+			else if (IsCitusLocalTable(distributedTableId))
+			{
+				hasPostgresLocalTable = true;
```

Comment:
this seems a bit incorrect, if we only have citus local tables then I would say we don't have postgres local tables. Should we rename this variable?